lib/dao.js

```diff
@@ -1477,25 +1496,34 @@ DataAccessObject.update =
 DataAccessObject.updateAll = function (where, data, options, cb) {
   if (stillConnecting(this.getDataSource(), this, arguments)) return;
 
-  assert(arguments.length >= 2, 'At least two arguments are required');
-  if (options === undefined && cb === undefined) {
+  assert(arguments.length >= 1, 'At least one argument is required');
+
+  if (data === undefined && options === undefined && cb === undefined && arguments.length === 1) {
```

Comment:
> Cool...  I think you'd agree that this particular method *is* a special
case.... most methods don't have the possible side effect of overwriting
your entire dataset if the arguments are misinterpreted...

Yes, definitely.